util/src/main/java/org/datacommons/util/McfParser.java

```diff
@@ -36,21 +35,21 @@
   private int curEntityLineIdx = 0;
   private String prevEntity;
   private boolean finished = false;
+  private Iterator<String> lines;
 
   // Create an McfParser instance based on type and a bool indicating whether the MCF is resolved
   // (DCIDs assigned).
-  public static McfParser init(Mcf.McfType type, boolean isResolved) {
-    McfParser parser = new McfParser();
-    parser.graph = McfGraph.newBuilder();
-    parser.graph.setType(type);
-    parser.isResolved = isResolved;
-    parser.curEntity = "";
-    parser.prevEntity = "";
+  public static McfParser init(Mcf.McfType type, String fileName, boolean isResolved)
+      throws IOException {
+    McfParser parser = init(type, isResolved);
+    parser.lines =
+        Files.lines(FileSystems.getDefault().getPath(fileName), StandardCharsets.UTF_8).iterator();
     return parser;
   }
 
   // Parse a string with instance nodes in MCF format into the McfGraph proto.
-  public static McfGraph parseInstanceMcfString(String mcf_string, boolean isResolved) {
+  public static McfGraph parseInstanceMcfString(String mcf_string, boolean isResolved)
```

Comment:
javastyle :) mcfString or mcfStr
```suggestion
  public static McfGraph parseInstanceMcfString(String mcfString, boolean isResolved)
```

Reply:
Done